overlord/snapstate/snapstate.go

```diff
@@ -2374,11 +2374,8 @@ func ConfigDefaults(st *state.State, deviceCtx DeviceContext, snapName string) (
 		return nil, state.ErrNoState
 	}
 
-	constraints := &gadget.ModelConstraints{
-		Classic: release.OnClassic,
-	}
-
-	gadgetInfo, err := gadget.ReadInfo(info.MountDir(), constraints)
+	// at this point we don't need constraints checks, they happend earlier
```

Comment:
There's a typo here:
```suggestion
	// at this point we don't need constraints checks, they happened earlier
```